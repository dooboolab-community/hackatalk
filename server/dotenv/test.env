JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
