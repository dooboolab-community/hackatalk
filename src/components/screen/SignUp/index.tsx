import * as Yup from 'yup';

import { Formik, FormikActions, FormikProps } from 'formik';
import {
  FormikInput,
  Props,
  SignUpFormValues,
  StyledButtonWrapper,
  StyledForm,
  StyledSafeAreaView,
  StyledScrollView,
} from './styles';
import React, { ReactElement, memo } from 'react';

import { Alert } from 'react-native';
import Button from '../../shared/Button';
import { getString } from '../../../../STRINGS';
import { withTheme } from 'styled-components/native';

const signUpValidationSchema = Yup.object({
  email: Yup.string()
    .email(getString('EMAIL_FORMAT_NOT_VALID'))
    .required(getString('EMAIL_REQUIRED')),
  password1: Yup.string()
    .min(6, getString('PASSWORD_MIN'))
    .required(getString('PASSWORD_REQUIRED')),
  password2: Yup.string()
    .min(6, getString('PASSWORD_MIN'))
    .oneOf([Yup.ref('password1'), null], getString('PASSWORD_MUST_MATCH'))
    .required(getString('PASSWORD_REQUIRED')),
  name: Yup.string()
    .min(3, getString('NAME_MIN'))
    .required(getString('NAME_REQUIRED')),
  status: Yup.string(),
});

const onSignUpSubmit = (
  {
    email,
    password1,
    name,
    status,
  }: SignUpFormValues,
  {
    resetForm,
    setSubmitting,
  }: FormikActions<SignUpFormValues>,
): void => {
  Alert.alert(
    'Signed Up',
    `You've signed up with 
      email: ${email},
      password: ${password1},
      name: ${name},
      status: ${status}
      successfully!
    `
  );
  setSubmitting(false);
  resetForm();
};

const getLabelName = (key: string): string => {
  const newKey = key.toUpperCase();
  if (key.startsWith('password')) {
    return `${getString(
      newKey.substring(
        0,
        newKey.length - 1
      )
    )}${newKey.substring(
      newKey.length - 1,
      newKey.length
    )}`;
  }
  return getString(newKey);
};

const getInputType = (key: string): string => {
  if (key.startsWith('password')) {
    return 'password';
  } else if (key === 'email') {
    return 'email';
  }
  return 'text';
};

function SignUpPage({
  theme,
  screenProps: {
    changeTheme,
  },
}: Props): ReactElement {
  return (
    <StyledSafeAreaView>
      <StyledScrollView>
        <Formik
          initialValues={{
            email: '',
            password1: '',
            password2: '',
            name: '',
            status: '',
          }}
          validationSchema={signUpValidationSchema}
          onSubmit={(values, actions): void => onSignUpSubmit(values, actions)}
        >
          {({ isValid, values, handleSubmit }: FormikProps<SignUpFormValues>): ReactElement => (
            <StyledForm testID="formTest">
              {Object.keys(values).map((key: string) => (
                <FormikInput
                  key={key}
                  label={key}
                  name={key}
                  type={getInputType(key)}
                  placeholder={getLabelName(key)}
                  placeholderTextColor={theme.placeholder}
                />
              ))}
              <StyledButtonWrapper>
                <Button
                  testID="btnSignUpConfirm"
                  isDisabled={!isValid}
                  style={{ alignSelf: 'flex-end' }}
                  onPress={handleSubmit}
                >
                  {getString('REGISTER')}
                </Button>
              </StyledButtonWrapper>
            </StyledForm>
          )}
        </Formik>
      </StyledScrollView>
    </StyledSafeAreaView>
  );
};
SignUpPage.displayName = 'SignUp';

export default memo(withTheme(SignUpPage));
