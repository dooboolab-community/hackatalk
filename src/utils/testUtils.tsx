import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
