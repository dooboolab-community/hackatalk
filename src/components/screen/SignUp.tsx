import {
  AuthPayload,
  MUTATION_SEND_VERIFICATION,
  MUTATION_SIGN_UP,
  MutationSendVerificationInput,
  MutationSignUpInput,
} from '../../graphql/mutations';
import { Button, EditText } from '@dooboo-ui/native';
import React, { ReactElement, useState } from 'react';
import { showAlertForGrpahqlError, validateEmail, validatePassword } from '../../utils/common';

import { AuthStackNavigationProps } from '../navigation/AuthStackNavigator';
import { ScrollView } from 'react-native-gesture-handler';
import StatusBar from '../shared/StatusBar';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useMutation } from '@apollo/react-hooks';
import { useThemeContext } from '@dooboo-ui/native-theme';

const Container = styled.SafeAreaView`
  flex: 1;
  background: ${({ theme }): string => theme.background};
`;

const Wrapper = styled.View`
  margin: 44px;
`;

const ButtonWrapper = styled.View`
  width: 100%;
  margin-top: 32px;
  flex-direction: row-reverse;
`;

interface Props {
  navigation: AuthStackNavigationProps<'SignUp'>;
}

function Page(props: Props): ReactElement {
  const { navigation } = props;
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');
  const [name, setName] = useState<string>('');
  const [statusMessage, setStatusMessage] = useState<string>('');

  const [errorEmail, setErrorEmail] = useState<string>('');
  const [errorPassword, setErrorPassword] = useState<string>('');
  const [errorConfirmPassword, setErrorConfirmPassword] = useState<string>('');
  const [errorName, setErrorName] = useState<string>('');
  const [signingUp, setSigningUp] = useState<boolean>(false);

  const { theme } = useThemeContext();
  const [signUp] = useMutation<{ signUp: AuthPayload }, MutationSignUpInput>(MUTATION_SIGN_UP);
  const [sendVerification] =
    useMutation<{ sendVerification: boolean }, MutationSendVerificationInput>(MUTATION_SEND_VERIFICATION);

  const requestSignUp = async (): Promise<void> => {
    if (!validateEmail(email) || !validatePassword(password) || name.length < 2 || password !== confirmPassword) {
      if (!validateEmail(email)) {
        setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));
      }
      if (!validatePassword(password)) {
        setErrorPassword(getString('PASSWORD_MIN'));
      }
      if (name.length < 2) {
        setErrorName(getString('NAME_MIN'));
      }
      if (password !== confirmPassword) {
        setErrorConfirmPassword(getString('PASSWORD_MUST_MATCH'));
      }

      return;
    }

    setSigningUp(true);
    const variables = {
      user: {
        email,
        name,
        password,
        statusMessage,
      },
    };

    try {
      const { data } = await signUp({ variables });
      const { data: emailVerificationData } = await sendVerification({
        variables: {
          email,
        },
      });
      if (emailVerificationData?.sendVerification) {
        // email sent
      }
      const user = data?.signUp.user;

      if (user) {
        navigation.replace('VerifyEmail', {
          email,
        });
      }
    } catch ({ graphQLErrors }) {
      showAlertForGrpahqlError(graphQLErrors);
    } finally {
      setSigningUp(false);
    }
  };

  return (
    <Container>
      <StatusBar />
      <ScrollView style={{ alignSelf: 'stretch' }}>
        <Wrapper>
          <EditText
            testID="input-email"
            errorTestID="error-email"
            textStyle={{
              color: theme.fontColor,
            }}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            label={getString('EMAIL')}
            placeholder="hello@example.com"
            value={email}
            onChangeText={(text: string): void => {
              setEmail(text);
              setErrorEmail('');
            }}
            errorText={errorEmail}
            onSubmitEditing={requestSignUp}
          />
          <EditText
            testID="input-password"
            errorTestID="error-password"
            textStyle={{
              color: theme.fontColor,
            }}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            placeholder="********"
            label={getString('PASSWORD')}
            value={password}
            onChangeText={(text: string): void => {
              setPassword(text);
              setErrorPassword('');
            }}
            style={{ marginTop: 32 }}
            errorText={errorPassword}
            onSubmitEditing={requestSignUp}
            secureTextEntry={true}
          />
          <EditText
            testID="input-confirm-password"
            errorTestID="error-confirm-password"
            textStyle={{
              color: theme.fontColor,
            }}
            placeholder="********"
            label={getString('CONFIRM_PASSWORD')}
            value={confirmPassword}
            onChangeText={(text: string): void => {
              setConfirmPassword(text);
              setErrorConfirmPassword('');
            }}
            style={{ marginTop: 32 }}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            errorText={errorConfirmPassword}
            onSubmitEditing={requestSignUp}
            secureTextEntry={true}
          />
          <EditText
            testID="input-name"
            errorTestID="error-name"
            textStyle={{
              color: theme.fontColor,
            }}
            label={getString('NAME')}
            placeholder={getString('NAME_HINT')}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            value={name}
            onChangeText={(text: string): void => {
              setName(text);
              setErrorName('');
            }}
            style={{ marginTop: 32 }}
            errorText={errorName}
            onSubmitEditing={requestSignUp}
          />
          <EditText
            testID="input-status"
            errorTestID="error-status"
            textStyle={{
              color: theme.fontColor,
            }}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            label={getString('STATUS')}
            placeholder={getString('STATUS_MSG_HINT')}
            value={statusMessage}
            onChangeText={(text: string): void => {
              setStatusMessage(text);
            }}
            style={{ marginTop: 32 }}
            onSubmitEditing={requestSignUp}
          />
          <ButtonWrapper>
            <Button
              testID="btn-sign-up"
              isLoading={signingUp}
              onPress={requestSignUp}
              containerStyle={{ height: 52, width: '50%', backgroundColor: theme.btnPrimary }}
              textStyle={{ color: theme.btnPrimaryFont, fontSize: 16 }}
              text={getString('SIGN_UP')}
            />
          </ButtonWrapper>
        </Wrapper>
      </ScrollView>
    </Container>
  );
}

export default Page;
