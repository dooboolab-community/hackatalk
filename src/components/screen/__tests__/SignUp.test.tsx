import 'react-native';

import React, { ReactElement } from 'react';
import {
  RenderResult,
  act,
  cleanup,
  fireEvent,
  render,
  wait,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import SignUp from '../SignUp';
import renderer from 'react-test-renderer';

let props: any;
let component: ReactElement;

describe('[SignUp] rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<SignUp {...props} />);
  });

  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[SignUp] interaction', () => {
  let testingLib: RenderResult;

  beforeAll(() => {
    props = createTestProps();
    component = createTestElement(<SignUp {...props} />);
    testingLib = render(component);
  });

  it('should invoke changeText event handler when email changed', async () => {
    const textInput = testingLib.getByTestId('input-email');
    await wait(() => expect(textInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(textInput, 'email@email.com');
    });

    expect(textInput.props.value).toEqual('email@email.com');
  });

  it('should invoke changeText event handler when password changed', async () => {
    testingLib = render(component);
    const textInput = testingLib.getByTestId('input-password');
    await wait(() => expect(textInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(textInput, 'pw test');
    });
    expect(textInput.props.value).toEqual('pw test');
  });

  it('should confirm password', async () => {
    testingLib = render(component);
    const textInput = testingLib.getByTestId('input-confirm-password');
    await wait(() => expect(textInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(textInput, 'pw test');
    });
    expect(textInput.props.value).toEqual('pw test');
  });

  it('should invoke changeText event when name changed', async () => {
    const textInput = testingLib.getByTestId('input-name');
    await wait(() => expect(textInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(textInput, 'dooboo');
    });

    expect(textInput.props.value).toEqual('dooboo');
  });

  it('should invoke changeText event when status changed', async () => {
    const textInput = testingLib.getByTestId('input-status');
    await wait(() => expect(textInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(textInput, 'I am good');
    });

    expect(textInput.props.value).toEqual('I am good');
  });

  describe('onSignUp', () => {
    beforeAll(() => {
      props = createTestProps();
      component = createTestElement(<SignUp {...props} />);
      testingLib = render(component);
    });

    describe('check validation', () => {
      beforeAll(() => {
        props = createTestProps();
        component = createTestElement(<SignUp {...props} />);
        testingLib = render(component);
      });

      it('should show error text when the email is not validated', async () => {
        const emailInput = testingLib.getByTestId('input-email');
        await wait(() => expect(emailInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(emailInput, 'email@email');
        });

        const passwordInput = testingLib.getByTestId('input-password');
        await wait(() => expect(emailInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(passwordInput, 'Abc123##');
        });

        const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
        await wait(() => expect(confirmPasswordInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(confirmPasswordInput, 'Abc123##');
        });

        const nameInput = testingLib.getByTestId('input-name');
        await wait(() => expect(nameInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(nameInput, 'dooboo');
        });

        const btnSignUp = testingLib.getByTestId('btn-sign-up');
        await wait(() => expect(btnSignUp).toBeTruthy());
        act(() => {
          fireEvent.press(btnSignUp);
        });

        const errorText = testingLib.getByTestId('error-email');
        await wait(() => expect(errorText).toBeTruthy());
      });

      it('should show error text when the password is not validated', async () => {
        const emailInput = testingLib.getByTestId('input-email');
        await wait(() => expect(emailInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(emailInput, 'email@email.com');
        });

        const passwordInput = testingLib.getByTestId('input-password');
        await wait(() => expect(emailInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(passwordInput, 'Abc');
        });

        const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
        await wait(() => expect(confirmPasswordInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(confirmPasswordInput, 'Abc');
        });

        const nameInput = testingLib.getByTestId('input-name');
        await wait(() => expect(nameInput).toBeTruthy());
        act(() => {
          fireEvent.changeText(nameInput, 'dooboo');
        });

        const btnSignUp = testingLib.getByTestId('btn-sign-up');
        await wait(() => expect(btnSignUp).toBeTruthy());
        act(() => {
          fireEvent.press(btnSignUp);
        });

        const errorText = testingLib.getByTestId('error-password');
        await wait(() => expect(errorText).toBeTruthy());
      });
    });

    it('should show error text when the name is not validated', async () => {
      const emailInput = testingLib.getByTestId('input-email');
      await wait(() => expect(emailInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(emailInput, 'email@email.com');
      });

      const passwordInput = testingLib.getByTestId('input-password');
      await wait(() => expect(emailInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(passwordInput, 'Abc123##');
      });

      const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
      await wait(() => expect(confirmPasswordInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(confirmPasswordInput, 'Abc123##');
      });

      const nameInput = testingLib.getByTestId('input-name');
      await wait(() => expect(nameInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(nameInput, 'me');
      });

      const btnSignUp = testingLib.getByTestId('btn-sign-up');
      await wait(() => expect(btnSignUp).toBeTruthy());
      act(() => {
        fireEvent.press(btnSignUp);
      });

      const errorText = testingLib.getByTestId('error-name');
      await wait(() => expect(errorText).toBeTruthy());
    });

    it('should show error text when the password is not confirmed', async () => {
      const emailInput = testingLib.getByTestId('input-email');
      await wait(() => expect(emailInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(emailInput, 'email@email.com');
      });

      const passwordInput = testingLib.getByTestId('input-password');
      await wait(() => expect(emailInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(passwordInput, 'Abc123##');
      });

      const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
      await wait(() => expect(confirmPasswordInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(confirmPasswordInput, 'Adjdj123??');
      });

      const nameInput = testingLib.getByTestId('input-name');
      await wait(() => expect(nameInput).toBeTruthy());
      act(() => {
        fireEvent.changeText(nameInput, 'dooboo');
      });

      const btnSignUp = testingLib.getByTestId('btn-sign-up');
      await wait(() => expect(btnSignUp).toBeTruthy());
      act(() => {
        fireEvent.press(btnSignUp);
      });

      const errorText = testingLib.getByTestId('error-confirm-password');
      await wait(() => expect(errorText).toBeTruthy());
    });
  });

  it('should call signUp when button has clicked and navigate to SignIn', async () => {
    const emailInput = testingLib.getByTestId('input-email');
    await wait(() => expect(emailInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(emailInput, 'email@email.com');
    });

    const passwordInput = testingLib.getByTestId('input-password');
    await wait(() => expect(passwordInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(passwordInput, 'Abc123##');
    });

    const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
    await wait(() => expect(confirmPasswordInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(confirmPasswordInput, 'Abc123##');
    });

    const nameInput = testingLib.getByTestId('input-name');
    await wait(() => expect(nameInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(nameInput, 'dooboo');
    });

    const btnSignUp = testingLib.getByTestId('btn-sign-up');
    await wait(() => expect(btnSignUp).toBeTruthy());

    jest.useFakeTimers();
    act(() => {
      fireEvent.press(btnSignUp);
      jest.runAllTimers();
    });

    await act(() => wait());
    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);
  });

  it('should do nothing when there is no navigation', async () => {
    props = createTestProps({
      navigation: null,
    });
    component = createTestElement(<SignUp {...props} />);
    testingLib = render(component);

    const emailInput = testingLib.getByTestId('input-email');
    await wait(() => expect(emailInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(emailInput, 'email@email.com');
    });

    const passwordInput = testingLib.getByTestId('input-password');
    await wait(() => expect(passwordInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(passwordInput, 'Abc123##');
    });

    const confirmPasswordInput = testingLib.getByTestId('input-confirm-password');
    await wait(() => expect(confirmPasswordInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(confirmPasswordInput, 'Abc123##');
    });

    const nameInput = testingLib.getByTestId('input-name');
    await wait(() => expect(nameInput).toBeTruthy());

    act(() => {
      fireEvent.changeText(nameInput, 'dooboo');
    });

    const btnSignUp = testingLib.getByTestId('btn-sign-up');
    await wait(() => expect(btnSignUp).toBeTruthy());

    jest.useFakeTimers();
    act(() => {
      fireEvent.press(btnSignUp);
      jest.runAllTimers();
    });

    await act(() => wait());
    expect(props.navigation).toBeNull();
  });

  afterAll((done) => {
    cleanup();
    done();
  });
});
