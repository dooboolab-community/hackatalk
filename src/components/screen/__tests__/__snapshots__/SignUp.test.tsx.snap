// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SignUp] rendering test renders as expected 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <ScrollView
        collapsable={false}
        style={
          Object {
            "alignSelf": "stretch",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "marginBottom": 44,
                "marginLeft": 44,
                "marginRight": 44,
                "marginTop": 44,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  null,
                ]
              }
            >
              Email
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    },
                    Object {
                      "borderBottomWidth": 0.6,
                      "borderColor": "#2E2E2E",
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  placeholder="hello@example.com"
                  placeholderTextColor="#676c7a"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "color": "#2c374e",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 15,
                        "fontWeight": "500",
                        "height": "100%",
                        "paddingBottom": 15,
                        "paddingTop": 15,
                      },
                      Object {
                        "color": "#151A25",
                      },
                    ]
                  }
                  testID="input-email"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  null,
                ]
              }
            >
              Password
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    },
                    Object {
                      "borderBottomWidth": 0.6,
                      "borderColor": "#2E2E2E",
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  placeholder="********"
                  placeholderTextColor="#676c7a"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "color": "#2c374e",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 15,
                        "fontWeight": "500",
                        "height": "100%",
                        "paddingBottom": 15,
                        "paddingTop": 15,
                      },
                      Object {
                        "color": "#151A25",
                      },
                    ]
                  }
                  testID="input-password"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  null,
                ]
              }
            >
              Confirm password
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    },
                    Object {
                      "borderBottomWidth": 0.6,
                      "borderColor": "#2E2E2E",
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  placeholder="********"
                  placeholderTextColor="#676c7a"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "color": "#2c374e",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 15,
                        "fontWeight": "500",
                        "height": "100%",
                        "paddingBottom": 15,
                        "paddingTop": 15,
                      },
                      Object {
                        "color": "#151A25",
                      },
                    ]
                  }
                  testID="input-confirm-password"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  null,
                ]
              }
            >
              Name
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    },
                    Object {
                      "borderBottomWidth": 0.6,
                      "borderColor": "#2E2E2E",
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  placeholder="Please write your name"
                  placeholderTextColor="#676c7a"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "color": "#2c374e",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 15,
                        "fontWeight": "500",
                        "height": "100%",
                        "paddingBottom": 15,
                        "paddingTop": 15,
                      },
                      Object {
                        "color": "#151A25",
                      },
                    ]
                  }
                  testID="input-name"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  null,
                ]
              }
            >
              Status
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    },
                    Object {
                      "borderBottomWidth": 0.6,
                      "borderColor": "#2E2E2E",
                    },
                    undefined,
                    null,
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  placeholder="Please write your status message"
                  placeholderTextColor="#676c7a"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "color": "#2c374e",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontSize": 15,
                        "fontWeight": "500",
                        "height": "100%",
                        "paddingBottom": 15,
                        "paddingTop": 15,
                      },
                      Object {
                        "color": "#151A25",
                      },
                    ]
                  }
                  testID="input-status"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row-reverse",
                  "marginTop": 32,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              delayPressIn={30}
              style={
                Object {
                  "backgroundColor": "#1E6EFA",
                  "height": 52,
                  "width": "50%",
                }
              }
              testID="btn-sign-up"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "borderColor": "blue",
                      "flexDirection": "row",
                      "height": 52,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#069ccd",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#ffffff",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Sign Up
                </Text>
              </View>
            </View>
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;
